Use a drop down instead of a button set for selecting locale

lalitpatel opened the pull request:
### Summary of changes:
* Use a drop down instead of a button set for selecting locale.
* Add an option to submit translations
* Add translations for en_US 😁 
* Open external links in a new tab.

https://user-images.githubusercontent.com/13775/110975272-89c1ca80-8385-11eb-8c35-4ad4853b9ad6.mov

HSZemi commented:
Do we _really_ need an en_US language option, which essentially means we have to do each change twice? :grimacing: 

What is the reasoning behind adding `target="_blank"` to the links in the footer? Using `target="_blank"` on links just because they are external is a generally [discouraged practice](https://css-tricks.com/use-target_blank/) afaik.

The Language Switcher looks good :+1:  
Maybe add to the README that whoever submits a new language must volunteer to keep it up to date in the future as well :grin:

lalitpatel commented:
> Do we _really_ need an en_US language option, which essentially means we have to do each change twice? 😬

You are right. I didn't consider this while adding the language. I added it just for the sake of it. No real good reason :) 

> What is the reasoning behind adding `target="_blank"` to the links in the footer? Using `target="_blank"` on links just because they are external is a generally [discouraged practice](https://css-tricks.com/use-target_blank/) afaik.

Primary reason was "Internal” links and “External” links https://css-tricks.com/use-target_blank/#a-bad-reason-internal-links-and-external-links-are-different . Intuitively, it made sense for me to open the link a new tab for "Submit a Translation". 

Personally, I usually open "further reading" type of links in a new tab so that it doesn't break the flow of the site that I am currently browsing and can go to those pages later. I have removed the `target=_blank` from footer links. I recommend we keep it for "Submit a translation" Link.

HSZemi commented on src/languages/de_DE.json:
```diff
@@ -142,7 +142,7 @@
     "VLD_907": "Das Preset enthält einen versteckten Zug, der nie aufgedeckt wird",
     "VLD_908": "Das Preset enthält keine Züge",
-    "VLD_909": "Die angegebene presetId ist ungültig"
+    "VLD_909": "Die angegebene preset id ist ungültig"
```
This is a technical "you are apparently fiddling with the application, but you're doing it wrong" error message, so I think keeping it as `presetId` here is better

lalitpatel replied:
Got it. Will revert.

HSZemi commented on src/containers/LocaleSelector.tsx:
```diff
@@ -0,0 +1,16 @@
+}
+
+export function mapDispatchToProps(dispatch: Dispatch<actions.Action>) {
+    return {};
```
I am by no means a React expert: Do we need this container, or can we just use the component directly?

lalitpatel replied:
Yes, we do not need this container. Removed it!

HSZemi reviewed (commented):
Sorry it took so long to have a proper look again :neutral_face:
